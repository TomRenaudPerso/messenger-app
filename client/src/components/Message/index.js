import React  from "react";
import { number, string } from "prop-types";
import {useSelector} from "react-redux";

const Message = ({ authorId, author, body }) => {
	const loggedUser = useSelector(state => state.loggedUser);
	const messageStyles = {
		display: 'flex',
		borderRadius: '1.875rem',
		padding: '0.625rem',
		margin: '0.625rem',
		width: 'fit-content',
		maxWidth: '31.25rem',
		backgroundColor: loggedUser.id === authorId ? '#006AFF' : 'lightGrey',
		color: loggedUser.id === authorId ? 'white' : 'black',
		fontWeight: 400,
	};
	const containerMessageStyles = {
		display: 'flex',
		flexDirection: 'column',
		margin: '0.625rem 0'
	};
	const authorStyles = {
		marginLeft: '1.25rem',
		color: 'grey',
	}

	return (
		<div style={containerMessageStyles}>
			{ loggedUser.id !== authorId && <span data-testid="authorMessage" style={authorStyles}>{author}</span> }
			<div data-testid="bodyMessage" style={messageStyles}>
				{body}
			</div>
		</div>
	);
}

Message.propTypes = {
	authorId: number.isRequired,
	author: string.isRequired,
	body: string.isRequired
}

export default Message;