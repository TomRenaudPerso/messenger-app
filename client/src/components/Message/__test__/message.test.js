import React from "react";
import { render } from "@testing-library/react";
import Message from "../index";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

beforeEach(() => {
	Object.defineProperty(window, 'matchMedia', {
		writable: true,
		value: jest.fn().mockImplementation(query => ({
			matches: false,
			media: query,
			onchange: null,
			addListener: jest.fn(), // deprecated
			removeListener: jest.fn(), // deprecated
			addEventListener: jest.fn(),
			removeEventListener: jest.fn(),
			dispatchEvent: jest.fn(),
		})),
	});
});

const initialState = {
	loggedUser: {
		"id": 1,
		"nickname": "Thibaut",
		"token": "xxxx"
	}
};
const mockStore = configureStore();
let store;

test('check content and author', () => {
	store = mockStore(initialState);
	const { getByTestId } = render(
		<Provider store={store}>
			<Message
				authorId={2}
				author="John Doe"
				body="Hello World"
			/>
		</Provider>
	);
	const body = getByTestId("bodyMessage");
	const author = getByTestId("authorMessage");
	expect(body.textContent).toBe("Hello World");
	expect(author.textContent).toBe("John Doe");
});

test("should match snapshot", () => {
	store = mockStore(initialState);
	const tree = renderer.create(
		<Provider store={store}>
			<Message
				authorId={2}
				author="John Doe"
				body="Hello World"
			/>
		</Provider>
	).toJSON();
	expect(tree).toMatchSnapshot();
});