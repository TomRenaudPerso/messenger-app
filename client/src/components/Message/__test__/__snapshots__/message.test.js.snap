// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "margin": "0.625rem 0",
    }
  }
>
  <span
    data-testid="authorMessage"
    style={
      Object {
        "color": "grey",
        "marginLeft": "1.25rem",
      }
    }
  >
    John Doe
  </span>
  <div
    data-testid="bodyMessage"
    style={
      Object {
        "backgroundColor": "lightGrey",
        "borderRadius": "1.875rem",
        "color": "black",
        "display": "flex",
        "fontWeight": 400,
        "margin": "0.625rem",
        "maxWidth": "31.25rem",
        "padding": "0.625rem",
        "width": "fit-content",
      }
    }
  >
    Hello World
  </div>
</div>
`;
